src/operator/src/statement.rs

```diff
@@ -104,14 +100,12 @@ impl StatementExecutor {
 
     pub async fn execute_sql(&self, stmt: Statement, query_ctx: QueryContextRef) -> Result<Output> {
         match stmt {
-            Statement::Query(_) | Statement::Explain(_) => {
+            Statement::Query(_) | Statement::Explain(_) | Statement::Delete(_) => {
                 self.plan_exec(QueryStatement::Sql(stmt), query_ctx).await
             }
 
             Statement::Insert(insert) => self.insert(insert, query_ctx).await,
```

Comment:
Why `Insert` and `Delete` are on different paths

Reply:
There is a fast path for insert. Delete doesn't have this fast path and can only use "plan + exec" like query.